robottelo/datafactory.py

```diff
@@ -288,3 +289,51 @@ def valid_usernames_list():
         gen_string("alphanumeric"),
         gen_string("numeric"),
     ]
+
+
+def valid_http_credentials(url_encoded=None):
+    """Returns a list of valid credentials for HTTP authentication
+
+    :param url_encoded: flag for quoting special characters
+    :return: A list of dictionaries with user and password credentials
+    """
+    credentials = [
+        {u'login': 'admin', u'pass': 'changeme'},
+        {u'login': '@dmin', u'pass': 'changeme'},
+        {u'login': 'adm/n', u'pass': 'changeme'},
+        {u'login': 'admin2', u'pass': 'ch@ngeme'},
+        {u'login': 'admin3', u'pass': 'chan:eme'},
+        {u'login': 'admin4', u'pass': 'chan/eme'},
+        {u'login': 'admin5', u'pass': 'ch@n:eme'},
+        {u'login': '0', u'pass': 'mypassword'},
+        {u'login': '0123456789012345678901234567890123456789',
+         u'pass': 'changeme'},
+    ]
+    if url_encoded is True:
+        return [{u'login': quote(cred['login'], ''),
```

Comment:
@sthirugn exactly
`urllib2.quote`'s 2nd parameter stands for `safe characters` - characters that should be excluded from quoting and by default it is equal to `'/'`. As I want to include `/` in quoting, i need to pass my custom safe "list", which is empty.
Anyway due to missing `urllib2` dependency I'm going to use `urllib` as this is already in requirements and this one contains `quote_plus` which converts that automatically so no safe list is required